scripts/kind-build-test.sh

```diff
@@ -0,0 +1,154 @@
+}
+
+function clean_up {
+    if [[ "$PRESERVE" == false ]]; then
+        "${SCRIPT_PATH}"/delete-kind-cluster.sh $DELETE_CLUSTER_ARGS || :
+        return
+    fi
+    echo "To resume test with the same cluster use: \"-c $TMP_DIR\""""
+}
+
+function exit_and_fail {
+    local pod_id=$(get_nth_worker_pod || :)
+    kubectl logs "${pod_id}" --namespace kube-system || :
+    END=$(date +%s)
+    echo "⏰ Took $(expr "${END}" - "${START}")sec"
+    echo "❌ NTH Integration Test FAILED $CLUSTER_NAME! ❌"
+    exit 1
+}
+
+function get_nth_worker_pod {
+    kubectl get pods -n kube-system \
+      --selector 'app.kubernetes.io/name=aws-node-termination-handler' \
+      --field-selector="spec.nodeName=$CLUSTER_NAME-worker,status.phase=Running" \
+      --sort-by=.metadata.creationTimestamp \
+      --output jsonpath='{.items[-1].metadata.name}'
+}
+
+USAGE=$(cat << 'EOM'
+  Usage: bash run.sh [-p] [-s] [-o] [-b <TEST_BASE_NAME>] [-c <CLUSTER_CONTEXT_DIR>] [-i <AWS Docker image name>] [-s] [-v K8S_VERSION]
+  Executes a test within a provisioned kubernetes cluster with NTH and IMDS pre-loaded.
+
+  Example: bash run.sh -p -s petstore
+
+          Optional:
+            -b          Base name of test (will be used for cluster too)
+            -c          Cluster context directory, if operating on an existing cluster
+            -p          Preserve kind k8s cluster for inspection
+            -i          Provide AWS Service docker image
+            -s          Provide AWS Service name (ecr, sns, sqs, petstore, bookstore)
+            -o          Override path w/ your own kubectl and kind binaries
+            -v          Kubernetes Version (Default: 1.16) [1.14, 1.15, 1.16, 1.17, and 1.18]
+
+EOM
+)
+
+# Process our input arguments
+while getopts "ps:ioc:b:v:" opt; do
+  case ${opt} in
+    p ) # PRESERVE K8s Cluster
+        echo "❄️  This run will preserve the cluster as you requested"
+        PRESERVE=true
+      ;;
+    s ) # AWS Service name
+        echo "Running Docker build as you requested for ${OPTARG} service"
+        AWS_SERVICE=$(echo "${OPTARG}" | tr '[:upper:]' '[:lower:]')
+        echo $AWS_SERVICE
+      ;;
+    i ) # AWS Service Docker Image
+        AWS_SERVICE_DOCKER_IMG="${OPTARG}"
+      ;;
+    o ) # Override path with your own kubectl and kind binaries
+        DELETE_CLUSTER_ARGS="${DELETE_CLUSTER_ARGS} -o"
+        PROVISION_CLUSTER_ARGS="${PROVISION_CLUSTER_ARGS} -o"
+        OVERRIDE_PATH=1
+      ;;
+    c ) # Cluster context directory to operate on existing cluster
+        TMP_DIR="${OPTARG}"
+      ;;
+    b ) # Base cluster name
+        CLUSTER_NAME_BASE="${OPTARG}"
+      ;;
+    v ) # K8s VERSION
+        K8S_VERSION="${OPTARG}"
+      ;;
+    \? )
+        echo "${USAGE}" 1>&2
+        exit
+      ;;
+  esac
+done
+
+if [ -z $TMP_DIR ]; then
+    TMP_DIR=$("${SCRIPT_PATH}"/provision-kind-cluster.sh -b "${CLUSTER_NAME_BASE}" -v "${K8S_VERSION}" "${PROVISION_CLUSTER_ARGS}")
```

Comment:
I think eventually I want to replace `provision-kind-cluster.sh` scritpt with a Bash function `up-kind-cluster` that functions more like this: https://github.com/jaypipes/aws-controllers-k8s/blob/master/scripts/lib/cluster.sh#L14